### PR TITLE
Remove instance during deprovisioning in application broker

### PR DESCRIPTION
**Pull request by adamwalach**
**Description**

Changes proposed in this pull request:

- during deprovisioning remove instance entity from storage 

**Related issue(s)**
* #546

### CODE REVIEW

File: components/application-broker/internal/storage/driver/memory/entity_operation.go [hunk 1]
```diff
@@ -101,10 +110,27 @@ func (s *InstanceOperation) GetAll(iID internal.InstanceID) ([]*internal.Instanc
 	for i := range opsForInstance {
 		out = append(out, opsForInstance[i])
 	}
-
+	sort.Sort(instanceOperations(out))
```
Comment (jasiu001) [inline components/application-broker/internal/storage/driver/memory/entity_operation.go]: Add some single unit test for sorting

File: components/application-broker/internal/storage/testing/operation_test.go [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+		"A1": {"iID-01", "oID-01", "state01",  "pHash-01"},
+		"A2": {"iID-03", "oID-02", "state02", "pHash-02"},
+		"A3": {"iID-03", "oID-03", "state03",  "pHash-03"},
+		"B3": {"iID-03", "oID-04", "state04", "pHash-04"},
+	} {
+		i := &internal.InstanceOperation{
+			OperationID:     internal.OperationID(ft.oID),
+			InstanceID:            internal.InstanceID(ft.iID),
+			State: internal.OperationState(ft.oState),
+			ParamsHash:    ft.pHash,
```
Comment (jasiu001) [inline components/application-broker/internal/storage/testing/operation_test.go]: Use go fmt or before_commit :)

Comment (adamwalach) [reply components/application-broker/internal/storage/testing/operation_test.go]: Formatting fixed

File: components/application-broker/internal/storage/driver/memory/entity_operation.go (hunk 1 above)
Comment (adamwalach) [reply components/application-broker/internal/storage/driver/memory/entity_operation.go]: I implemented two tests TestInstanceOperationFindLast and TestInstanceOperationGetAllSortOrder

**Pull request merged**